src/v/cloud_roles/refresh_credentials.h

```diff
@@ -109,6 +110,8 @@ class refresh_credentials {
 
         uint16_t api_port() const { return _api_port; }
 
+        s3::aws_region_name region() const { return _region; }
```

Comment:
since this PR is all green field, it seems like this addition of region support can be folded back into the commits that introduced refresh_credentials